### PR TITLE
Facility and filters data format change

### PR DESCRIPTION
**Pull request by JonAlexandra**
This changes the data format such that we should be able to handle it almost no matter what they come up with. The PR is huge and messy and will probably introduce some bugs.

### CODE REVIEW

File: src/components/FacilityCard.vue [hunk 1]
```diff
@@ -69,12 +73,13 @@
...
           <a
-            :href="`https://maps.apple.com/?daddr=${facility.city}+${facility.zip}+${facility.address}&dirflg=d`"
+            :href="`https://maps.apple.com/?daddr=${facility.address.city}+${facility.address.zip}+${facility.address.street}&dirflg=d`"
             target="_blank"
             class="icon-link"
             aria-label="Naviger til faciliteten"
           >
-            {{ facility.address }}, {{ facility.city }}, {{ facility.zip }}
+            {{ facility.address.street }}, {{ facility.address.city }},
+            {{ facility.address.zip }}
```
Comment (MagnusHBrunbjerg) [inline src/components/FacilityCard.vue:82]: Don't know why the format is like this but normally it's done like in the suggestion

```suggestion
            {{ facility.address.street }}, {{ facility.address.zip }} {{ facility.address.city }}
```

**Review by MagnusHBrunbjerg — CHANGES_REQUESTED** [review]

**Review by MagnusHBrunbjerg — APPROVED** [review]

File: src/components/FacilityCard.vue (hunk 1 above)
Comment (JonAlexandra) [reply src/components/FacilityCard.vue:82]: This is the prettier formatting in relation to long lines. It is ugly agreed, but we should keep following the auto formatting.

File: src/util/search-util.ts [hunk 2]
```diff
@@ -30,58 +18,83 @@ const searchKeys = [
...
   }
 
   return facilities
-    .filter(facility => facilityScores[facility.uId] > 0)
-    .sort((a, b) => facilityScores[b.uId] - facilityScores[a.uId])
+    .filter(facility => facilityScores[facility.uuid] > 0)
+    .sort((a, b) => facilityScores[b.uuid] - facilityScores[a.uuid])
+}
+
+function searchAndScoreFilters(
```
[the remark this answers is not in the document]
Comment (JonAlexandra) [reply src/util/search-util.ts:74]: Good catch

File: src/util/search-util.ts [hunk 3]
```diff
@@ -30,58 +18,83 @@ const searchKeys = [
...
-                text,
-                searchTheshold
-              )
+          for (const itemValue of Object.values(value)) {
+            if (!itemValue) {
+              break
             }
+
+            facilityScores[facility.uuid] += scoreString(
+              itemValue.toString().toLowerCase().trim(),
```
[the remark this answers is not in the document]
Comment (JonAlexandra) [reply src/util/search-util.ts:52]: Agreed

File: src/stores/types.ts [hunk 4]
```diff
@@ -1,17 +1,36 @@
...
+  lonLatPosition: [lon: number, lat: number]
+  address: {
+    street: string
+    zip: string
+    city: string
+  }
+
+  // These last ones will certainly need changing, but will wait,
+  // until more UI decisions have been made
+  contactInfo: {
```
[the remark this answers is not in the document]
Comment (JonAlexandra) [reply src/stores/types.ts:23]: Agreed. Remember to mention this tomorrow.

**Review by MagnusHBrunbjerg — APPROVED** [review]
LGTM

**Pull request merged**